app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export)
+    ActiveRecord::Base.connection_pool.with_connection do
+      create_export_folder(csv_export)
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # TODO delete files with no record associations
+    CsvExport.old.each do |csv_export|
+      csv_export.destroy
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.status! :started
+    deploy_csv_export(csv_export)
+    notify_of_creation(csv_export)
+  rescue NoMethodError, IOError, ActiveRecord::ActiveRecordError => e
```

Comment:
I think I eliminated all cases where calling a method on nil would occur.  The report has to pull data from several linked models.  I don't think it would ever come up since I replaced the known cases with try, but it may still occur.  

If it gets eliminated and not caught the job would simply fail with no cleanup, and the user won't get the report because it would never reach a finished state.  The user may wonder why the report is taking so long, however.  

The no method error was coming up because of deleted projects, stages or users.  I think now with the dependent delete/destroys it wouldn't appear for newer records, but with some of the older historical records it may come up.